server/game/gameSystems/DiscardHandSystem.ts

```diff
@@ -0,0 +1,32 @@
+import { EventName, Location } from '../core/Constants';
+import { AbilityContext } from '../core/ability/AbilityContext';
+import * as Helpers from '../core/utils/Helpers.js';
+import type Player from '../core/Player';
+import { IPlayerTargetSystemProperties, PlayerTargetSystem } from '../core/gameSystem/PlayerTargetSystem';
+
+// eslint-disable-next-line @typescript-eslint/no-empty-object-type
+export interface IDiscardHandSystemProperties extends IPlayerTargetSystemProperties {}
+
+/**
+ * A {@link GameSystem} which discards a player's entire hand
+ */
+export class DiscardHandSystem<TContext extends AbilityContext = AbilityContext> extends PlayerTargetSystem<TContext, IDiscardHandSystemProperties> {
+    public override name = 'discardHand';
+    public override eventName = EventName.OnHandDiscarded;
+    public override readonly costDescription: string = 'discard hand';
+    public override readonly effectDescription: string = 'discard hand';
+
+    public override eventHandler(event): void {
+        const players = Helpers.asArray(event.player) as Player[];
+
+        players.forEach((player) =>
+            player
+                .getCardPile(Location.Hand)
+                .forEach((card) => player.moveCard(card, Location.Discard))
```

Comment:
We just settled on a pattern for how we want to handle an effect that discards multiple cards, check out DiscardCardsFromHand.generateEventsForCards. We emit one individual discard event for each card and then an aggregate event indicating whether the entire discard effect resolved or not. This makes it possible for the discarding of individual cards to be reacted to much more easily.